java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -1576,6 +1576,25 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_bitwiseMergeAndSetValidit
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_copyWithValidity(
+    JNIEnv *env, jobject j_object, jlong exemplar_handle, jlong validity_column_handle) {
+  JNI_NULL_CHECK(env, exemplar_handle, "ColumnView handle is null", 0);
+  JNI_NULL_CHECK(env, validity_column_handle, "Validity column handle is null", 0);
+  try {
+    cudf::jni::auto_set_device(env);
+
+    auto const exemplar = *reinterpret_cast<cudf::column_view *>(exemplar_handle);
+    auto const validity = *reinterpret_cast<cudf::column_view *>(validity_column_handle);
+    if (exemplar.size() != validity.size()) {
+      JNI_THROW_NEW(env, cudf::jni::ILLEGAL_ARG_CLASS,
+                    "Exemplar and validity columns must have the same size", 0);
```

Comment:
`new_column_with_validity` is already checking for this (and more), so it seems redundant here.

Reply:
Should I keep the Java checks instead of the C++ ones? Perhaps an `IllegalArgumentException` is preferable in this case?

Reply:
In practice I'm not sure it makes much difference.  It's redundant to check in both places, IMO.

Reply:
> It's redundant to check in both places, IMO.

Agreed.

Reply:
Removed the redundant error checking. Kept the error handling in the CUDA layer.